docs/kb/semgrep-ci/bitbucket-jenkins-freestyle-projects.md

```diff
@@ -6,48 +6,59 @@ tags:
-1. In **Triggers > Select an Action** select **Created**, **Updated** and **Push**.
+1. In the **Build Triggers** section, click **<i class="fa-solid fa-square-check"></i> Build with Bitbucket Push and Pull Request Plugin**. 
+1. In **Triggers > Select an Action** select **Created**, **Updated**, and **Push**.
 ![Build triggers for the freestyle project](/img/kb/bitbucket-jenkins-freestyle-events.png)
-1. In the **Build environment** section declare the `SEMGREP_APP_TOKEN` selecting the option **Use secret text or file.**
+1. In the **Build environment** section, declare the `SEMGREP_APP_TOKEN` by selecting **Use secret text or file.** Set **Variable** to `SEMGREP_APP_TOKEN` and **Credentials > Specific credentials** to your token value. Click **Add** to save your changes.
 ![Filled-out the secrets text option](/img/kb/bitbucket-jenkins-freestyle-token.png)
 :::note
-The `SEMGREP_APP_TOKEN` must be defined as a credential in Jenkins settings.
+Ensure that you have [defined `SEMGREP_APP_TOKEN` as a credential](https://www.jenkins.io/doc/book/using/using-credentials/#configuring-credentials) in Jenkins.
 :::
 
-## Running full scans 
-In the section **Build Steps** add a new **Execute Shell** step with the logic explained in the steps below:
+## Run full scans 
+In the **Build Steps** section, add an **Execute Shell** step with the logic below:
 ```
+#!/bin/bash
+
+REPO_URL=$GIT_URL
+REPO_NAME=$(echo "$GIT_URL" | awk -F'/' '{print $(NF-1)"/"$(NF)}' | sed 's/.git$//')
+
 docker run \
     -e SEMGREP_APP_TOKEN=$SEMGREP_APP_TOKEN \
+    -e SEMGREP_REPO_URL=$REPO_URL \
+    -e SEMGREP_REPO_NAME=$REPO_NAME \
     -v "$(pwd):$(pwd)" --workdir $(pwd) \
     semgrep/semgrep semgrep ci
 ```
-Now, after a push to the main branch a new Semgrep full scan runs.
+:::note
+- The variable `SEMGREP_REPO_URL` is helpful to link the Semgrep findings with the repository.
+:::
+
+:::note
+- Setting the variable `SEMGREP_REPO_NAME` is a good practice as it allows to specify an accurate name to the Semgrep project.
+:::
+
+
+Now, a full scan runs when you push changes to the main branch.
 
-## Running pull requests scans (diff scans)
+## Running pull requests scans (diff-aware scans)
 
-For this approach, it is required to set some Environment variables that you would typically see natively in Bitbucket but not on the Jenkins side. For Semgrep to integrate natively, the following environment variables are needed:
-SEMGREP_REPO_URL
-SEMGREP_REPO_NAME
-SEMGREP_BRANCH
-SEMGREP_PR_ID
-SEMGREP_BASELINE_REF
+The diff-aware scan configuration uses a computed merge base. To achieve that it is needed to specify the source branch: `SEMGREP_BRANCH` and the target branch: `SEMGREP_BASELINE_REF`. In addition, setting the `SEMGREP_REPO_NAME` and `SEMGREP_PR_ID` allows Semgrep to identify the connected project and related PR.
```

Comment:
Actually, this diff-aware scan config doesn't use a computed merge base. It would be better if it did, but it might require further modifications to the script (I'm not exactly sure how deep the BB checkout is). If we retain the current script, this should look more like:

```suggestion
The diff-aware scan configuration must specify a merge base to compare the PR changes against. To achieve that, specify the pull request target branch as `SEMGREP_BASELINE_REF`, and set `SEMGREP_BRANCH` to the pull request source branch to ensure it's correctly identified. Set the `SEMGREP_REPO_NAME` as described above for full scans, and add `SEMGREP_PR_ID` so Semgrep can send comments to the related PR.
```